scripts/sbatch.py

```diff
@@ -0,0 +1,476 @@
+                    header_dict[header.value] = header.type(value)
+                    break
+                elif header.type is bool and "--{}".format(header.id) in line:
+                    header_dict[header.value] = True
+                    
+    return header_dict
+
+
+def process_mem_per_cpu(mem_per_cpu):
+    """Convert a mem_per_cpu string to float"""
+    if isinstance(mem_per_cpu, str):
+            value = float(re.findall(r"[-+]?\d*\.\d+|\d+", mem_per_cpu)[0])
+    else:
+        value = mem_per_cpu
+    if 'G' not in mem_per_cpu.upper():  # no unit (default =MB) or unit=MB, transforming to GB
+        value = float(value) / MB_PER_GB
+    
+    return value
+
+
+def process_header_values(slurm_header_dict, slurm_conf):
+    """
+       Process header dict values
+       get defaul values if not provided
+    """
+    for header in [SlurmHeader.ntasks, SlurmHeader.cpus_per_task, SlurmHeader.ntasks_per_core]:
+        if not slurm_header_dict[header.value]:
+            slurm_header_dict[header.value] = header.default
+            slurm_header_dict[header.value] = header.default
+        else:
+            slurm_header_dict[header.value] = int(slurm_header_dict[header.value])
+    #special treatment for WCT.
+    # convert str to datetime -> datetime to second
+    if not slurm_header_dict[SlurmHeader.mem_per_cpu.value]:
+        slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(get_value_from_conf(
+            slurm_conf,
+            slurm_header_dict.partition,
+            keyword="DefMemPerCPU",
+            all2float=True,
+        ))
+    else:
+        slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value]
+        )
+    #rule for share node resource
+    slurm_header_dict[SlurmHeader.exclusive.value]=shared_TRES_conf(slurm_conf, slurm_header_dict[SlurmHeader.partition.value]) 
+    return slurm_header_dict
+
+
+def get_hpc_conf(partition):
+    """Get hpc name and path to slurm.conf according to partition name in the slurm file header"""
+    for machine in [MauiPartition,MauiAncilPartition,MahuikaPartition]:
+        for p in machine:
+            if partition in p.id:
+                slurm_conf = p.config_path  
+                hpc = p.hpc
+                cpu_per_node = p.cpus
+                mem_per_node = p.mem
+            
+                return hpc, slurm_conf, cpu_per_node, mem_per_node
+    raise IndexError("no matching partition name found")
+
+
+def compare_hours_and_sbatch(requested_hours, available_hours, hpc, sl_to_sbatch, sbatch_extra_args):
+    """
+       Compare requested hour with available hours
+       and either submit the job or reject
+    """
+    if requested_hours <= available_hours:
+        cmd = "{} -M {} {} {}".format(SBATCH_BIN, hpc, sl_to_sbatch, sbatch_extra_args)
```

Comment:
" ".join(sbatch_extra_args)
Otherwise the list will be put into the string e.g. "['--ncores' '40']"